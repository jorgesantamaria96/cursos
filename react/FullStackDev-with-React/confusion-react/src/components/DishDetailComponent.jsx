import React from "react";
import {
  Card,
  CardImg,
  CardBody,
  CardText,
  CardTitle,
  Breadcrumb,
  BreadcrumbItem,
} from "reactstrap";
import { Link } from "react-router-dom";

const DishDetail = (props) => {
  const RenderDish = (dish) => {
    if (dish != null) {
      return (
        <Card>
          <CardImg top src={dish.dish.image} alt={dish.dish.name} />
          <CardBody>
            <CardTitle>{dish.dish.name}</CardTitle>
            <CardText>{dish.dish.description}</CardText>
          </CardBody>
        </Card>
      );
    }
  };

  const RenderComments = (comments) => {
    if (comments != null) {
      const comentarios = comments.dish.map((comment) => {
        return (
          <div key={comment.id}>
            <CardText>{comment.comment}</CardText>
            <CardText>
              -- {comment.author}, {comment.date}
            </CardText>
            <br />
          </div>
        );
      });

      return (
        <Card className="mb-5 ">
          <CardBody>
            <CardTitle>Comments</CardTitle>
            {comentarios}
          </CardBody>
        </Card>
      );
    }
    return <div />;
  };

  return (
    <div className="container">
      <div className="row">
        <Breadcrumb>
          <BreadcrumbItem>
            <Link to="/menu">Menu</Link>
          </BreadcrumbItem>
          <BreadcrumbItem>{props.dish.name}</BreadcrumbItem>
        </Breadcrumb>
        <div className="col-12">
          <h3>{props.dish.name}</h3>
          <hr />
        </div>
      </div>
      <div className="row">
        <div className="col-12 col-md-5 m-1">
          <RenderDish dish={props.dish} />
        </div>
        <div className="col-12 col-md-5 m-1">
          <RenderComments dish={props.comments} />
        </div>
      </div>
    </div>
  );
};

export default DishDetail;
