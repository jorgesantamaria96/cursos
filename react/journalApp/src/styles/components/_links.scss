.link {
  text-decoration: none;
  color: $dark-grey;
  margin-top: 20px;

  &:hover {
    text-decoration: underline;
  }
}
