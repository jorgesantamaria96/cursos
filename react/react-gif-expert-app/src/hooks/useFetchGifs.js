import { useEffect, useState } from "react";
import { getGifs } from "../helpers/getGifs";

export const useFetchGifs = (category) => {
  const [state, setState] = useState({
    data: [],
    loading: true,
  });

  useEffect(() => {
    getGifs(category).then((images) => {
      setState({
        loading: false,
        data: images,
      });
    });
  }, [category]);

  return state;
};
