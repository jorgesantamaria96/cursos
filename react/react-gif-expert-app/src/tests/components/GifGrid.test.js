import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";

describe("Pruebas en <GridGif />", () => {
  const category = "Goku";

  test("debe mostrar el componente correctamente", () => {
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("debe de mostrar items cuando se cargan las imagenes useFetchGifs", () => {

    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(true);
    expect(wrapper.find('GifGridItem').length).toBe(0);
  });
});
