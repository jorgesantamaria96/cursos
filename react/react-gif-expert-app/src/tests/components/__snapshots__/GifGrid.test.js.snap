// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GridGif /> debe de mostrar items cuando se cargan las imagenes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Goku
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="cardGrid"
  />
</Fragment>
`;

exports[`Pruebas en <GridGif /> debe mostrar el componente correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Goku
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="cardGrid"
  />
</Fragment>
`;
